src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/TrackingOptInModal.java

```diff
@@ -20,153 +21,161 @@
-        clickRejectButton();
-        isTrackingRequestsNotSend(urlPatterns, networkTrafficInterceptor);
+  public void isTrackingRequestSend(List<String> elementsList,
+                                    NetworkTrafficInterceptor networkTrafficInterceptor) {
+    for (int i = 0; i < elementsList.size(); i++) {
+      wait.forSuccessfulResponseByUrlPattern(networkTrafficInterceptor, elementsList.get(i));
     }
+  }
 
-    public void verifyTrackingRequestsSend(String[] instantGlobals, List<String> urlPatterns,
-                                                  NetworkTrafficInterceptor networkTrafficInterceptor,
-                                                  Page page){
-        getUrl(urlWithInstantGlobals(instantGlobals, page));
-        clickAcceptButton();
-        isTrackingRequestSend(urlPatterns, networkTrafficInterceptor);
-    }
-
-    public void verifyTrackingRequestsSend(List<String> urlPatterns,
-                                                 NetworkTrafficInterceptor networkTrafficInterceptor,
-                                                 Page page){
-        getUrl(urlBuilder.appendQueryStringToURL(urlBuilder.getUrlForPage(page), MODAL_INSTANT_GLOBAL));
-        clickAcceptButton();
-        isTrackingRequestSend(urlPatterns, networkTrafficInterceptor);
-    }
+  private boolean isSuccessfulResponseByUrlPattern(
+      final NetworkTrafficInterceptor trafficInterceptor,
+      final String pattern) {
+    try {
+      entry = trafficInterceptor.getEntryByUrlPattern(pattern);
+      return entry.getResponse().getStatus() < 400;
 
-    public String urlOptInModalDisplayedOasis(Page page){
-        return urlBuilder.appendQueryStringToURL(urlBuilder.getUrlForPage(page), MODAL_INSTANT_GLOBAL);
+    } catch (NullPointerException ex) {
+      return false;
     }
+  }
 
-    public void isTrackingRequestsNotSend(List<String> elementsList,
-                                          NetworkTrafficInterceptor networkTrafficInterceptor) {
-        wait.forX(WAITNG_TIME_FOR_ALL_REQUESTS);
-        for(int i=0; i<elementsList.size(); i++){
-            Assertion.assertFalse(isSuccessfulResponseByUrlPattern(networkTrafficInterceptor, elementsList.get(i)),
-                    "Request to " + elementsList.get(i) + " services was found");
-        }
-    }
+  private static String appendTrackingOptOutParameters(String url, String[] instantGlobals) {
+    UrlBuilder urlBuilder = new UrlBuilder();
 
-    public void isTrackingRequestSend(List<String> elementsList,
-                                      NetworkTrafficInterceptor networkTrafficInterceptor) {
-        for(int i=0; i<elementsList.size(); i++){
-            wait.forSuccessfulResponseByUrlPattern(networkTrafficInterceptor,elementsList.get(i));
-        }
+    for (String instantGlobal : instantGlobals) {
+      url = urlBuilder.globallyEnableGeoInstantGlobalOnPage(url, instantGlobal);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Introduce a new variable instead of reusing the parameter "url". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1226)